### PR TITLE
attempt to suppress the WARN thatpops when loading an identity as a file

### PR DESCRIPTION
**Pull request by dovholuknf**

### CODE REVIEW

File: library/config.c [hunk 1]
```diff
@@ -50,12 +50,25 @@ int ziti_load_config(ziti_config *cfg, const char* cfgstr) {
     if (!cfgstr) {
         return ZITI_INVALID_CONFIG;
     }
+    bool seems_like_json = false;
+    const char* c = cfgstr;
+    while (*cfgstr && isspace((unsigned char)*cfgstr)) {
+        c++;
+    }
+    if (strncmp(c,"{",1) == 0) {
```
Comment (ekoby) [inline library/config.c:58]: `if (*c  == '{')` ??

Comment (dovholuknf) [reply library/config.c]: yep. done

File: library/config.c [hunk 2]
```diff
@@ -50,12 +50,25 @@ int ziti_load_config(ziti_config *cfg, const char* cfgstr) {
     if (!cfgstr) {
         return ZITI_INVALID_CONFIG;
     }
+    bool seems_like_json = false;
+    const char* c = cfgstr;
+    while (*cfgstr && isspace((unsigned char)*cfgstr)) {
```
Comment (ekoby) [inline library/config.c:55]: this should check `*c` not `cfgstr` otherwise it is infinite loop

Comment (dovholuknf) [reply library/config.c:55]: yeah, just so happened that the first char wasn't whitespace when i tested it. nice catch

File: library/config.c [hunk 3]
```diff
@@ -50,12 +51,25 @@ int ziti_load_config(ziti_config *cfg, const char* cfgstr) {
...
+        c++;
+    }
+    if (*c == '{') {
+        seems_like_json = true;
+    }
 
     memset(cfg, 0, sizeof(*cfg));
-    int rc = parse_ziti_config(cfg, cfgstr, strlen(cfgstr));
+    int rc;
+    if(seems_like_json) {
```
Comment (scareything) [inline library/config.c:65]: whitespace

**Review by ekoby — APPROVED** [review]

**Pull request merged**